frontend/ui/widget/textviewer.lua

```diff
@@ -214,6 +219,69 @@ function TextViewer:init(reinit)
     end
 
     -- buttons
+    local pin_buttons = self.file and
+        {
+            {
+                text = _("Pin current page"),
+                callback = function()
+                    if self.pinned_page == nil then
+                        local pin_button = self.button_table:getButtonById("go_to_pin")
+                        pin_button:enable()
+                        pin_button:refresh()
+                    end
+                    self.pinned_page = self.scroll_text_w.text_widget:getCharPosAtXY(0, 0)
+                    local pinned_pages = G_reader_settings:readSetting("textviewer_pinned_pages", {})
```

Comment:
What this is is actually a `pinned_char`.
I get that what's facing the user can be "pinned page" for consistency with Reader.
But it feels a bit odd to carry this lying name all around in the code and the setting name.
If you really want to keep using pinned_page, may be a comment somewhere saying "this is actually the index of the char at the top of the view/page to remember" or something.